src/services/hana.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from collections.abc import Generator
+
+from hdbcli import dbapi
+
+from utils.settings import (
+    DATABASE_PASSWORD,
+    DATABASE_PORT,
+    DATABASE_URL,
+    DATABASE_USER,
+)
+from utils.singleton_meta import SingletonMeta
+
+
+class Hana(metaclass=SingletonMeta):
+    """
+    Manages a singleton connection to the Hana database.
+
+    This class establishes a connection to the Hana database using the provided
+    credentials and configuration. It ensures that only one instance of the connection
+    exists at any time by utilizing the SingletonMeta metaclass.
+    """
+
+    connection: dbapi.Connection | None = None
+
+    def set_connection(self, connection: dbapi.Connection) -> None:
+        """
+        sets the database connection for the service.
+
+        args:
+            connection (dbapi.connection): the database connection object.
+        """
+        self.connection = connection
+
+
+def get_hana_connection() -> Generator[Hana, None, None]:
```

Comment:
What is the purpose of `Generator` here?

Reply:
It is no longer needed. I restructured the code but overlooked that this would make the generators obsolete. The original intention to close connections in a `finally` block. But since that is not done in any other place where we open connections, I adapted to that.